frontend/src/screens/wallet/send/Onchain.tsx

```diff
@@ -0,0 +1,234 @@
+import React from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import FormattedFiatAmount from "src/components/FormattedFiatAmount";
+import Loading from "src/components/Loading";
+import { LinkButton } from "src/components/ui/button";
+import { Input } from "src/components/ui/input";
+import { Label } from "src/components/ui/label";
+import { LoadingButton } from "src/components/ui/loading-button";
+import { Switch } from "src/components/ui/switch";
+import { useToast } from "src/components/ui/use-toast";
+import { ONCHAIN_DUST_SATS } from "src/constants";
+import { useBalances } from "src/hooks/useBalances";
+import { useMempoolApi } from "src/hooks/useMempoolApi";
+import { useSwapFees } from "src/hooks/useSwaps";
+import { RedeemOnchainFundsResponse, SwapResponse } from "src/types";
+import { request } from "src/utils/request";
+
+export default function Onchain() {
+  const { state } = useLocation();
+  const navigate = useNavigate();
+  const { toast } = useToast();
+  const { data: balances } = useBalances();
+  const { data: swapFees } = useSwapFees("out");
+  const { data: recommendedFees, error: mempoolError } = useMempoolApi<{
+    fastestFee: number;
+    halfHourFee: number;
+    economyFee: number;
+    minimumFee: number;
+  }>("/v1/fees/recommended");
+
+  const address = state?.args?.address as string;
+  const [isSwap, setSwap] = React.useState(false);
+  const [amount, setAmount] = React.useState("");
+  const [feeRate, setFeeRate] = React.useState("");
+  const [isLoading, setLoading] = React.useState(false);
+
+  React.useEffect(() => {
+    if (recommendedFees?.fastestFee) {
+      setFeeRate(recommendedFees.fastestFee.toString());
+    }
+  }, [recommendedFees]);
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      if (!balances) {
+        return;
+      }
+      if (!address) {
+        throw new Error("no BTC address set");
+      }
+      if (balances.onchain.spendable <= ONCHAIN_DUST_SATS) {
+        throw new Error(
+          "You currently don't have enough sats to pay for an onchain transaction. Consider swapping from Spending Balance."
+        );
+      }
+      setLoading(true);
+      if (isSwap) {
+        const swapOutResponse = await request<SwapResponse>("/api/swaps/out", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            swapAmount: +amount,
```

Comment:
I think we have a problem here: I wanted to send 100K sats but I sent 98K sats. I think in this case the fee should be applied on top

Reply:
If you pay for a product or service this won't work at all because you have to send **exactly** the correct amount to the destination address